lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts

```diff
@@ -38,7 +38,7 @@ import {
 export const ManagedIdentityUserAssignedIdQueryParameterNames = {
     MANAGED_IDENTITY_CLIENT_ID: "client_id",
     MANAGED_IDENTITY_OBJECT_ID: "object_id",
-    MANAGED_IDENTITY_RESOURCE_ID: "mi_res_id",
+    MANAGED_IDENTITY_RESOURCE_ID: "msi_res_id",
```

Comment:
The bug states that this change should only be done for IMDS, not for all sources.

Reply:
Also, pls take the time to improve your tests. The fact that you make a change like this and no test breaks is sign that code coverage can be improved.

Reply:
Isn't this change desirable to [more than just IMDS](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4911#issuecomment-2573920867)?

Reply:
It seems that only IMDS documents this as "msi_res_id" and the rest do not. App service does not. Not sure about the others.

Reply:
Actually, for "the others", only Machine Learning is applicable, and we assume it follows IMDS's "msi_res_id", too.